src/parser/jobs/mch/modules/Gauge.js

```diff
@@ -0,0 +1,195 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React, {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import Color from 'color'
+import TimeLineChart from 'components/ui/TimeLineChart'
+
+// Constants
+const MAX_GAUGE = 100
+
+const GCD_HEAT_GAIN = 5
+const BARREL_STABILIZER_HEAT_GAIN = 50
+const HYPERCHARGE_HEAT_COST = 50
+
+const CLEAN_SHOT_BATTERY_GAIN = 10
+const AIR_ANCHOR_BATTERY_GAIN = 20
+
+export default class Gauge extends Module {
+	static handle = 'gauge'
+	static title = t('mch.gauge.title')`Heat & Battery Gauge`
+	static dependencies = [
+		'brokenLog',
+		'suggestions',
+	]
+
+	_gauge = {
+		heat: {
+			current: 0,
+			waste: 0,
+			history: [],
+		},
+		battery: {
+			current: 0,
+			waste: 0,
+			history: [],
+		},
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('combo', {by: 'player', abilityId: [ACTIONS.HEATED_SLUG_SHOT.id, ACTIONS.HEATED_CLEAN_SHOT.id]}, this._onHeatGcd)
+		this.addHook('cast', {by: 'player', abilityId: [ACTIONS.HEATED_SPLIT_SHOT.id, ACTIONS.SPREAD_SHOT.id]}, this._onHeatGcd)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.BARREL_STABILIZER.id}, this._onStabilizer)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.HYPERCHARGE.id}, this._onOverheat)
+
+		this.addHook('combo', {by: 'player', abilityId: ACTIONS.HEATED_CLEAN_SHOT.id}, this._onCleanShot)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.AIR_ANCHOR.id}, this._onAirAnchor)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.AUTOMATON_QUEEN.id}, this._onQueen)
+
+		this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_pushToHistory(type) {
+		const timestamp = this.parser.currentTimestamp - this.parser.fight.start_time
+		this._gauge[type].history.push({t: timestamp, y: this._gauge[type].current})
+	}
+
+	_addGauge(type, amount) {
+		const gauge = this._gauge[type]
+		gauge.current += amount
+		if (gauge.current >= MAX_GAUGE) {
+			gauge.waste += (gauge.current - MAX_GAUGE)
+			gauge.current = MAX_GAUGE
+		}
+
+		this._pushToHistory(type)
+	}
+
+	_onHeatGcd() {
+		this._addGauge('heat', GCD_HEAT_GAIN)
```

Comment:
Feel free to use inline arrows for super short event handlers like this. Stuff like

`this.addHook('type', {filter:stuff}, () => this._addGauge('foo', bar))`

Reply:
Fair, I can tidy up this and `_onStabilizer()`.